### PR TITLE
Feature/api has auth

### PR DESCRIPTION
**Pull request by knordstrom**

### CODE REVIEW

File: api/employees.py [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
 
 from library.workday import Workday
 
 route = APIRouter(tags=["Employees"])
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="auth/login/openapi")
 
 @route.put('/employees')
-async def add_employees_from_workday_export(workday_csv_export: UploadFile) -> ApiResponse[str]:  
+async def add_employees_from_workday_export(me: Annotated[User, Depends(AuthManager.get_user_dependency(oauth2_scheme))],
+                                            workday_csv_export: UploadFile) -> ApiResponse[str]:
```
Comment (mshash8) [inline api/employees.py:23]: It's cool to see the "Depends" mechanism which helps us ensure that only authenticated users can access these endpoints.

File: library/data/external/gsuite.py [hunk 2]
```diff
@@ -139,41 +145,56 @@ def meet_spaces_client(self) -> meet_v2.SpacesServiceAsyncClient:
...
 
-    def __init__(self, email, creds):
-        self.email = email
+    def __init__(self, user: User, creds: str):
+        self.user = user
+        self.email = user.email
         self.creds = creds
+        self.is_local = os.getenv("IS_LOCAL") in ["True", "true", "1"]
+
+    def get_existing_redentials(self) -> Credentials:
```
Comment (mshash8) [inline library/data/external/gsuite.py]: I think this function might be missing a "c" for credentials.

File: library/data/external/gsuite.py [hunk 3]
```diff
@@ -139,41 +145,56 @@ def meet_spaces_client(self) -> meet_v2.SpacesServiceAsyncClient:
...
-        if os.path.exists(self.TOKEN_FILE):
-            creds = Credentials.from_authorized_user_file(self.TOKEN_FILE, self.SCOPES)
+        creds: Credentials = self.get_existing_redentials()
 
         # if there are no (valid) credentials availablle, let the user log in.
         if not creds or not creds.valid:
             if creds and creds.expired and creds.refresh_token:
                 creds.refresh(Request())
-            else:
+            elif self.is_local:
```
Comment (mshash8) [inline library/data/external/gsuite.py:173]: Is this to differentiate if the auth is happening from a local/prod environment?

File: pyproject.toml [hunk 4]
```diff
@@ -38,7 +38,6 @@ groq = "^0.5.0"
 langchain-groq = "^0.1.3"
 python-docx = "^1.1.2"
 pypdf = "^4.2.0"
-textextract = "^1.2.0"
 textract = "^1.6.5"
```
Comment (mshash8) [inline pyproject.toml:41]: Thank you!

File: library/data/external/gsuite.py (hunk 3 above)
Comment (knordstrom) [reply library/data/external/gsuite.py:173]: Yep. On local, we wtill need the browser auth to work on the api

**Pull request merged**